linera-core/src/updater.rs

```diff
@@ -621,31 +628,35 @@ where
         chain_id: ChainId,
         target_block_height: BlockHeight,
         delivery: CrossChainMessageDelivery,
+        latest_certificate: Option<GenericCertificate<ConfirmedBlock>>,
```

Comment:
This is basically an alternative to `target_block_height`, is it? I wonder whether we should use `Either`, and specify exactly one of them.